File: python/sdk/recapture_detect_demo.py
# -*- coding:utf-8 -*-
from ais_sdk.gettoken import get_token
from ais_sdk.utils import encode_to_base64
from ais_sdk.recapture_detect import recapture_detect
from ais_sdk.recapture_detect import recapture_detect_aksk

if __name__ == '__main__':
    #
    # access image recapture detect ,post data by token
    #
    user_name = '******'
    password = '******'
    account_name = '******'  # the same as user_name in commonly use

    demo_data_url = 'https://ais-sample-data.obs.myhwclouds.com/tagging-normal.jpg'
    token = get_token(user_name, password, account_name)

    # call interface use the url (token, image, url, threshold=0.95, scene=None)
    result = recapture_detect(token, "", demo_data_url, 0.75, ["recapture"])
    print result

    #
    # access image recapture detect ,post data by aksk
    #
    app_key = "*************"
    app_secret = "************"

    # call interface use the file
    result = recapture_detect_aksk(app_key, app_secret, encode_to_base64('data/recapture-detect-demo-1.jpg'), "", 0.75, ["recapture"])
    print result

    demo_data_url = 'https://ais-sample-data.obs.myhwclouds.com/tagging-normal.jpg'

    # call interface use the url (token, image, url, threshold=0.95, scene=None)
    result = recapture_detect_aksk(app_key, app_secret, "", demo_data_url, 0.75, ["recapture"])
    print result

File: javascript/recapture_detect_demo.js
/**
 * 翻拍识别服务的使用示例
 */
var recapture = require("./ais_sdk/recapture_detect");
var token = require("./ais_sdk/gettoken");
var utils = require("./ais_sdk/utils");

var username = "*******";       // 配置用户名
var domainname = "*******";     // 配置用户名
var password = "*******";       // 密码
var regionName = "cn-north-1";  // 配置区域信息

var filepath = "./data/recapture-detect-demo.jpg";
var data = utils.changeFileToBase64(filepath);

/**
 * token 方式获取结果
 * @type {string}
 */
demo_data_url = "https://ais-sample-data.obs.myhwclouds.com/recapture-detect.jpg";
token.getToken(username, domainname, password, regionName, function (token) {

    recapture.recapture_detect(token, data, "", 0.99, ['recapture'], function (result) {
        console.log(result);
    });

    recapture.recapture_detect(token, "", demo_data_url, 0.99, ['recapture'], function (result) {
        console.log(result);
    })
});

/**
 * aksk 方式获取结果
 * @type {string}
 */
var app_key = "*************";
var app_secret = "************";

recapture.recapture_detect_aksk(app_key, app_secret, data, "", 0.99, ['recapture'], function (result) {
    console.log(result);
});

recapture.recapture_detect_aksk(app_key, app_secret, "", demo_data_url, 0.99, ['recapture'], function (result) {
    console.log(result);
});